cmd/kubelet/app/server.go

```diff
@@ -783,7 +783,11 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 	}
 
 	// If systemd is used, notify it that we have started
-	go daemon.SdNotify(false, "READY=1")
+	go func() {
+		if _, err := daemon.SdNotify(false, "READY=1\n"); err != nil {
+			klog.Errorf("Unable to send systemd daemon successful start message: %v\n", err)
+		}
+	}()
 
```

Comment:
If you really want to avoid code duplication and ensure that the same code will always be called in both places, I'd suggest to wrap it into a named function and call it in both places. BTW, this would help to call it asynchronously here and synchronously in apiserver code (see my comment below).

Reply:
@sologgfun ping

Reply:
It's still called asynchronously in both places.